src/java.base/share/classes/java/net/Socket.java

```diff
@@ -1108,6 +1125,16 @@ public int read(byte[] b, int off, int len) throws IOException {
                 throw e;
             }
         }
+
+        private int getSoTimeout() {
+            try {
+                return parent.getSoTimeout();
+            } catch (Throwable t) {
+                // ignored - avoiding exceptions in jfr event data gathering
```

Comment:
This should be SocketException, not Throwable. That said, I think it would be useful to know why the SocketReadEvent includes the timeout. Is this used to see If read durations are close to the timeout? I assume once this code is fixed to deal with the exceptional case that the need to include the timeout for the success case will mostly go away.

Reply:
No.  I think it's a relic from the distant past though.  I think the timeout field should be removed.  It's not used on SocketChannel at all, and it doesn't seem useful on Socket.

Reply:
Should we log an RFE to that effect?